Logger/LogFactory.cs

```diff
@@ -1,11 +1,33 @@
 ﻿namespace Logger
 {
-    public class LogFactory
+    using System;
+    using System.IO;
+
+    public static class LogFactory
     {
-        public BaseLogger CreateLogger(string className)
+        private static string? FilePath { get; set; }
+
+        public static FileLogger CreateLogger(string className)
```

Comment:
```suggestion
        public static FileLogger? CreateLogger(string className)
```
This will fix the warning CS8603 below.

Reply:
Also, this method body can be optionally simplified to:
```C#
            if (File.Exists(FilePath))
            {
                return new FileLogger(FilePath)
                { Name = className };
            }
            return null;
```